src/api/emit.js

```diff
@@ -10,11 +10,16 @@ const CustomEvent = ((Event) => {
 })(window.CustomEvent);
 
 function createCustomEvent(name, opts = {}) {
-  if (CustomEvent) {
-    return new CustomEvent(name, opts);
+  let e;
+  if (Event) {
+    e = new Event(name, opts);
```

Comment:
@bradleyayers - I'm not totally sure what you're saying here. 

Are you saying anything in opts we know the constructor doesn't use (currently!?), explicitly add as a property on the event (like we do for detail).

Or only pass opts to the constructor we know it will take? Or some other variation.